hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -145,12 +143,9 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     OmBucketInfo omBucketInfo = null;
     if (bucketInfo.getBucketLayout() == null || bucketInfo.getBucketLayout()
         .equals(BucketLayoutProto.LEGACY)) {
```

Comment:
Not understood why when it is legacy also we use default type here?

Reply:
Since we use enum in the proto, it will assign the default value automatically when the bucket layout is not specified or passed. It will use the first indexed enum value as default, which is LEGACY. We have added the null check for the safer side, which is actually not required. 

Supported values are only OBJECT_STORE and FILE_SYSTEM_OPTIMIZED. Client/User won't pass LEGACY so the LEGACY can occur when the argument is missing.

I think, [allowed values](https://github.com/apache/ozone/blob/master/hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/CreateBucketHandler.java#L56) has to be revisited. @JyotinderSingh can you please check it quickly by removing LEGACY from `AllowedBucketLayouts`. I think to raise a separate sub-task to address client side validation for the SUPPORTED_VALUES, if it requires more code changes.


@bharatviswa504 Does this make sense to you?

```
enum BucketLayoutProto {
    LEGACY = 1;
    FILE_SYSTEM_OPTIMIZED = 2;
    OBJECT_STORE = 3;
}
```

[OmClientProtocol.proto#L556 BucketLayoutProto](https://github.com/apache/ozone/blob/master/hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto#L556)

Reply:
I have removed LEGACY from `AllowedBucketLayouts` and ran a few tests that seemed to be using that value. I currently do not see any errors coming up. I am including this change in my patch.

Reply:
Sure, I'll see if there's any failing tests and make required changes